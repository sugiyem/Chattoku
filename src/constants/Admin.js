export const AdminPower = {
  DELETE_POSTS: "Delete Posts",
  BAN_USER: "Ban Users From Forum"
};
