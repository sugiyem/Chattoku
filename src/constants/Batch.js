export const MAXIMUM_BATCH_SIZE = 400;
