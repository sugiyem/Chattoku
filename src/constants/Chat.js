export const DEFAULT_AVATAR_URL =
  "https://firebasestorage.googleapis.com/v0/b/chattoku-e2672.appspot.com/" +
  "o/default-profile.png?alt=media&token=e05118be-75e9-42ed-8568-c42a3d34d6f1";

export const chatType = {
  PRIVATE_CHAT: 0,
  GROUP_CHAT: 1
};
