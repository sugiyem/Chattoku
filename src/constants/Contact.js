export const contactType = {
  USER: 0,
  GROUP: 1
};
