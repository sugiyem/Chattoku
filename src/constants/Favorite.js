export const favoriteType = {
  ANIME: 0,
  GENRE: 1
};
