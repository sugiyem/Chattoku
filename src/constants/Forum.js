export const renderType = {
  CREATE: 0,
  EDIT: 1
};

export const manageType = {
  BAN: 0,
  ADMIN: 1
};
