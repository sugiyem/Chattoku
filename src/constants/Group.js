export const groupListType = {
  GROUP: 0,
  GROUP_INVITATION: 1
};

export const groupMemberType = {
  OWNER: 0,
  MEMBER: 1,
  PENDING_MEMBER: 2,
  NON_MEMBER: 3
};
