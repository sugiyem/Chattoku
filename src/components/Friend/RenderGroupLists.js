import React, { useState } from "react";
import { StyleSheet, View } from "react-native";
import { ListItem } from "react-native-elements";
import {
  leaveGroup,
  acceptGroupInvitation,
  declineGroupInvitation
} from "../../services/Friend/HandleGroup";
import { contactType } from "../../constants/Contact";
import { groupListType } from "../../constants/Group";
import ContactBar from "./ContactBar";
import ContactButtonGroup from "./ContactButtonGroup";
import Caution from "../Miscellaneous/Caution";

export default RenderGroupLists = ({ type, item, navigation }) => {
  const [isExpanded, setIsExpanded] = useState(false);

  const buttonDetails = [];

  switch (type) {
    case groupListType.GROUP:
      buttonDetails.push(
        {
          title: "Detail",
          type: "ionicon",
          icon: "open-outline",
          color: "blue",
          onPress: (item) =>
            navigation.navigate("GroupInfo", { groupData: item })
        },
        {
          title: "Message",
          type: "material-community",
          icon: "message-processing-outline",
          color: "blue",
          onPress: (item) => {
            const data = {
              id: item.id,
              name: item.name,
              description: item.description,
              img: item.img
            };

            navigation.navigate("GroupChatList");
            navigation.navigate("GroupChatDetail", {
              groupData: data
            });
          }
        },
        {
          title: "Leave Group",
          type: "ionicon",
          icon: "exit-outline",
          color: "red",
          onPress: (item) =>
            Caution("You will leave this group", () => leaveGroup(item.id))
        }
      );
      break;

    case groupListType.GROUP_INVITATION:
      buttonDetails.push(
        {
          title: "Detail",
          type: "ionicon",
          icon: "open-outline",
          color: "blue",
          onPress: (item) =>
            navigation.navigate("GroupRequestInfo", { groupData: item })
        },
        {
          title: "Accept Invitation",
          type: "material",
          icon: "check",
          color: "green",
          onPress: (item) => acceptGroupInvitation(item.id)
        },
        {
          title: "Decline Invitation",
          type: "material",
          icon: "close",
          color: "red",
          onPress: (item) => declineGroupInvitation(item.id)
        }
      );
      break;
  }

  return (
    <ListItem.Accordion
      bottomDivider
      content={<ContactBar type={contactType.GROUP} item={item} />}
      isExpanded={isExpanded}
      onPress={() => setIsExpanded(!isExpanded)}
    >
      {isExpanded && (
        <ContactButtonGroup item={item} buttonDetails={buttonDetails} />
      )}
    </ListItem.Accordion>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: "stretch",
    margin: 10,
    padding: 5
  }
});
