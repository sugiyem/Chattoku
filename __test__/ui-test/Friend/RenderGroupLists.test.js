import React from "react";
import renderer from "react-test-renderer";
import RenderGroupLists from "../../../src/components/Friend/RenderGroupLists";
import { groupListType } from "../../../src/constants/Group";

const mockGroupInfo = {
  id: "Group-1",
  name: "Weebs",
  description: "weebs only",
  img: "weeb-img"
};

describe("Test Group List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <RenderGroupLists type={groupListType.GROUP} item={mockGroupInfo} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
