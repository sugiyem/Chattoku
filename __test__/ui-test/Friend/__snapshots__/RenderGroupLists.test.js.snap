// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Group List UI Renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#87CEFA",
          "borderBottomWidth": 1,
          "borderColor": "#011F5B",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginVertical": 5,
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "height": 50,
            "marginRight": 10,
            "width": 50,
          }
        }
        testID="image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 25,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={25}
            onLoad={[Function]}
            source={
              Object {
                "uri": "weeb-img",
              }
            }
            style={
              Object {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            transition={true}
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1D428A",
              "fontSize": 17,
            }
          }
          testID="name"
        >
          Weebs
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1D428A",
              "fontSize": 15,
            }
          }
          testID="info"
        >
          weebs only
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        collapsable={false}
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "maxHeight": "0%",
        "opacity": 0,
      }
    }
  />,
]
`;
